worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java

```diff
@@ -254,115 +254,43 @@ private boolean isLocationGoodForStanding(BlockVector3 location) {
-                    }
-                    // Not enough room, reset the free position
-                    if (bottomLimit != 1) {
-                        freeStart = -1;
-                    }
-                }
-            } else {
-                freeStart = -1;
-                lastState = true;
-            }
+            ++y;
         }
 
         return false;
     }
 
     @Override
     public boolean descendLevel() {
+        final World world = getWorld();
         final Location pos = getBlockLocation();
         final int x = pos.getBlockX();
-        int y = Math.max(0, pos.getBlockY() - 1);
+        int y = Math.max(world.getMinY(), pos.getBlockY() - 1);
         final int z = pos.getBlockZ();
-        final Extent world = pos.getExtent();
-
-        BlockMaterial initialMaterial = world.getBlock(BlockVector3.at(x, y, z)).getMaterial();
-
-        boolean lastState = initialMaterial.isMovementBlocker() && initialMaterial.isFullCube();
-
-        int maxY = world.getMaxY();
-        if (y <= 2) {
-            return false;
-        }
+        int yLessSearchHeight = y - WorldEdit.getInstance().getConfiguration().defaultVerticalHeight;
+        int minY = Math.min(world.getMinY() + 1, yLessSearchHeight);
 
-        double freeEnd = -1;
-        double height = 1.85;
-        for (int level = y + 1; level > 0; level--) {
-            BlockState state;
-            if (level >= maxY) {
-                state = BlockTypes.VOID_AIR.getDefaultState();
-            } else {
-                state = world.getBlock(BlockVector3.at(x, level, z));
+        while (y >= minY) {
+            if (isLocationGoodForStanding(BlockVector3.at(x, y, z))
+                    && trySetPosition(Vector3.at(x + 0.5, y, z + 0.5))) {
+                return true;
             }
-            BlockType type = state.getBlockType();
-            BlockMaterial material = type.getMaterial();
 
-            if (!material.isFullCube() || !material.isMovementBlocker()) {
-                if (!lastState) {
-                    lastState = BlockTypeUtil.centralTopLimit(state) != 0;
-                    continue;
-                }
-                if (freeEnd == -1) {
-                    freeEnd = level + BlockTypeUtil.centralBottomLimit(state);
-                } else {
-                    double topLimit = BlockTypeUtil.centralTopLimit(state);
-                    double freeStart = level + topLimit;
-                    double space = freeEnd - freeStart;
-                    if (space >= height) {
-                        trySetPosition(Vector3.at(x + 0.5, freeStart, z + 0.5));
-                        return true;
-                    }
-                    // Not enough room, reset the free position
-                    if (topLimit != 0) {
-                        freeEnd = -1;
-                    }
-                }
-            } else {
-                lastState = true;
-                freeEnd = -1;
-            }
+            --y;
         }
```

Comment:
presumably the large diff above is just using upstream's logic etc?

Reply:
Yeah, majority is upstream diff and removing Y0 in favor of getting the min world height using the API.